plugins/jobs/src/js/components/Breadcrumbs.js

```diff
@@ -6,57 +6,59 @@ import Breadcrumb from "#SRC/js/components/Breadcrumb";
 import BreadcrumbTextContent from "#SRC/js/components/BreadcrumbTextContent";
 import PageHeaderBreadcrumbs from "#SRC/js/components/PageHeaderBreadcrumbs";
 
-export default function Breadcrumbs({ item, children, states }) {
-  function getBreadcrumb(item, id = "", name = "") {
-    const isDetailPage = [...item.path, item.name].join(".") === id;
-    const link = isDetailPage ? `/jobs/detail/${id}` : `/jobs/overview/${id}`;
-
+export default function Breadcrumbs({ path, name, children, extra }) {
+  /**
+   * @param  {string} link
+   * @param  {string[]} pathname path including name (if given)
+   * @param  {JSX.Element} [extra=null] extra JSX Components to render
+   *
+   * @return {JSX.Element} Breadcrumb Part Component
+   */
+  function buildJSX(link, pathname, extra = null) {
     return (
-      <Breadcrumb key={id} title="Jobs">
+      <Breadcrumb key={pathname.join(".")} title="Jobs">
         <BreadcrumbTextContent>
-          <Link to={link}>{name === "" ? "Jobs" : name}</Link>
+          <Link to={link}>{pathname.slice(-1)}</Link>
         </BreadcrumbTextContent>
-
-        {isDetailPage ? states : null}
+        {extra}
       </Breadcrumb>
     );
   }
 
-  function getBreadcrumbList(item) {
-    if (item == null) {
-      return [];
-    }
-
-    const pathSegments = [...item.path, item.name];
-    const segments = pathSegments.map((_, index) =>
-      pathSegments.slice(0, index + 1)
-    );
+  // we're starting from the back, so first we have children
+  let breadcrumbs = React.Children.toArray(children);
 
-    return segments.map(segment =>
-      getBreadcrumb(item, segment.join("."), segment[segment.length - 1])
-    );
+  // if we got a name, thats a detail link
+  if (name !== undefined && name !== "") {
+    breadcrumbs = [
+      buildJSX(
+        ["/jobs/detail/"].concat(path, [name]).join("."),
+        path.concat([name]),
+        extra
+      )
+    ].concat(breadcrumbs);
   }
 
-  let breadcrumbs = [];
-
-  if (item) {
-    breadcrumbs = [].concat(
-      getBreadcrumb(item),
-      getBreadcrumbList(item),
-      React.Children.toArray(children)
-    );
+  // build up breadcrumb for path, starting from the end!
+  if (path !== undefined) {
+    for (let index = path.length; index > 0; index--) {
+      breadcrumbs = [
+        buildJSX(
+          ["/jobs/overview/"].concat(path.slice(0, index)).join("."),
+          path.slice(0, index)
+        )
+      ].concat(breadcrumbs);
+    }
   }
 
+  // add root element, here its named "Jobs"
+  breadcrumbs = [buildJSX("/jobs/overview/", ["Jobs"])].concat(breadcrumbs);
+
   return <PageHeaderBreadcrumbs iconID="jobs" breadcrumbs={breadcrumbs} />;
 }
 
 Breadcrumbs.propTypes = {
-  states: PropTypes.oneOfType([
-    PropTypes.element,
-    PropTypes.arrayOf(PropTypes.element)
-  ]),
-  item: PropTypes.shape({
-    name: PropTypes.string.isRequired,
-    path: PropTypes.arrayOf(PropTypes.string).isRequired
-  }).isRequired
+  extra: PropTypes.node,
```

Comment:
🎨 I would try to think of something with a more specific meaning than "extra".  I was not able to understand the difference of the Breadcrumbs children and the extra. At least I would make clear that those extra are nodes, extraChildren, extraContent....

Reply:
Second thing I struggled with 😅

Currently we use this to display additional information (Schedule and Status, which I find it already hard to find one name for) but in theory, this can be anything 🤷‍♂️